bundles/io/org.eclipse.smarthome.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/MqttBrokerConnection.java

```diff
@@ -229,6 +237,24 @@ public MqttBrokerConnection(String host, @Nullable Integer port, boolean secure,
         setReconnectStrategy(new PeriodicReconnectStrategy());
     }
 
+    /**
+     * Create a new MQTT client connection to a server with the given port, host and port.
+     *
+     * @param protocol The transport protocol
+     * @param host A host name or address
+     * @param port A port or null to select the default port for a secure or insecure connection
+     * @param secure A secure connection
+     * @param clientId Client id. Each client on a MQTT server has a unique client id. Sometimes client ids are
+     *            used for access restriction implementations.
+     *            If none is specified, a default is generated. The client id cannot be longer than 65535 characters.
+     * @throws IllegalArgumentException If the client id or port is not valid.
+     */
+    public MqttBrokerConnection(Protocol protocol, String host, @Nullable Integer port, boolean secure,
+            @Nullable String clientId) {
+        this(host, port, secure, clientId);
```

Comment:
* Make the protocol field final and remove the initialization on the declaration line.
* Move the implementation from the other constructor to this one.
* Add `this(Protocol.TCP, host, port, secure, clientId);` to the other constructor.